Reader.py

```diff
@@ -17,34 +16,28 @@ def write_to_file(self):
             json.dump(self.students, f, indent=4)
 
     def find(self):
-        self.students = self.read_from_file()
-
-        for student in self.students:
-            if student['Name'] == self.search:
-                count = student['Searches']
-                self.students.remove(student)
-                # contain = True
-                update_item = { "Name": self.search,"Searches": count + 1}
-                self.students.append(update_item)
-                # print('search update: ',student)
-                print('Current size of file: ',len(self.students))
-                self.students.sort(key=lambda k: k['Searches'], reverse=True)
-                # print('updated list of self.students', self.students)
-                break
-        else:
-            new = input('does not exist, enter yes or no ?')
-            # if new == 'yes':
-            #     student['Name'] = input('enter new name : ')
-            #     student['Searches'] = 0
-            #     self.students.remove(student)
-            #     self.students.append(student)
-
-        self.write_to_file()
-
-def call_reader():
-    search = input('enter your search')
-    stud = Reader(search)
-    stud.find()
+        while True:
+            search = input('find student: ')
+            self.students = self.read_from_file()
+            for student in self.students:
+                if student['Name'] == search:
+                    print(f'{search} exists. Total items {len(self.students)}')
+                    count = student['Searches']
+                    self.students.remove(student)
+                    # contain = True
+                    update_item = { "Name": search,"Searches": count + 1}
+                    self.students.append(update_item)
+                    self.students.sort(key=lambda k: k['Searches'], reverse=True)
```

Comment:
Sorting is an expensive operation (in terms of memory and time). Don't do it in multiple places. Its already in Writer... can remove from here.

Reply:
done